src/core/app-template.cc

```diff
@@ -56,6 +56,7 @@ seastar_options_from_config(app_template::config cfg) {
 
 app_template::seastar_options::seastar_options()
     : program_options::option_group(nullptr, "seastar")
+    , config_file(*this, "config-file", std::nullopt, "Path to seastar.conf file")
```

Comment:
We can put here the current default (`~/.config/seastar/seastar.conf`), so it appears in command line help and we also avoid the special case below.
The slight drawback is that we have to implement substituting `~`, but that is simple enough.

Reply:
Is there std::/boost:: tilde expansion? [It's not just](https://www.gnu.org/software/bash/manual/html_node/Tilde-Expansion.html) `s/~/${HOME}/` and I don't feel enough courage to implement it thoroughly.

Reply:
I don't know of any such facilities, and I don't think its worth it just for this. But please do mention in the option description that we do have a default.

Reply:
why do we need to implement the tilde expansion? 

we already have `std::string(home) + "/.config/seastar/seastar.conf"`. i think what is missing is to extract it into a static helper function, which returns an `optional<std::string>`, so the ctor of `seastar_options` can use it for the default value of `config_file`.

Reply:
Placing expanded home path into default value, makes the help text context dependant. A `--help` ran as `user1` will have incorrect value for `user2`.

Reply:
i see. we need to expand it when actually consuming it.

Reply:
Yes.